File: day4/main.py


def request_server() -> str:
    """A user is given a prompt for requesting a server."""
    server = input("Enter a new server name: ")
    return server

def manage_servers(server: str) -> list:
    """This function will manage a list of servers."""
    servers = []
    static_servers = ["darkstar", "ai-core", "omni-1"]
    for s in static_servers:
        servers.append(s)
    servers.append(server)
    return servers


def main():
    """This is my main function."""
    server = request_server()
    servers = manage_servers(server)
    print("Server List:")

    for i, server in enumerate(servers):
        print(f"[{i}] {server} is online!")

    

if __name__ == "__main__":
    main()

File: day4/main.go
package main

import (
	"fmt"
)

func requestServer() string {
	var s string
	fmt.Print("Enter a new server name: ")
	fmt.Scan(&s)
	return s
}

func manageServers(server string) (servers []string) {
	staticServers := []string{"darkstar", "ai-core", "omni-1"}
	for _, s := range staticServers {
		servers = append(servers, s)
	}
	servers = append(servers, server)
	return 
}

func main() {
	server := requestServer()
	servers := manageServers(server)
	fmt.Print("Server List:\n")

	for i, s := range servers {
		fmt.Printf("[%d] %s is online!\n", i, s)
	}
}